lib/vm/src/export.rs

```diff
@@ -52,7 +52,23 @@ pub struct VMFunction {
 
     /// A “reference” to the instance through the
     /// `InstanceRef`. `None` if it is a host function.
-    pub instance_ref: Option<InstanceRef>,
+    pub instance_ref: Option<WeakOrStrongInstanceRef>,
+}
+
+impl Clone for VMFunction {
+    fn clone(&self) -> Self {
+        Self {
+            address: self.address.clone(),
+            vmctx: self.vmctx.clone(),
+            signature: self.signature.clone(),
+            kind: self.kind.clone(),
+            call_trampoline: self.call_trampoline.clone(),
+            instance_ref: self
+                .instance_ref
+                .as_ref()
+                .map(|v| WeakOrStrongInstanceRef::Strong(v.get_strong().unwrap())),
```

Comment:
Cloning it automatically make ti strong?

Reply:
It's a bit weird this behavior. Perhaps we want to be more explicit having a custom method for this saying: `into_strong` that makes it strong.

Reply:
Yeah, was unsure to what extent we want to encapsulate that logic, but this is the secret sauce that makes this solution invisible to the user and behave correctly in every situation. Weak externs automatically become strong when they're shared (via cloning)